### PR TITLE
#160451493 Setup unit testing.

### PR DESCRIPTION
**Pull request by samdiano**
#### What does this PR do?
- This PR setup unit testing with mocha and chai
#### Description of Task to be completed?
-install mocha, chai, and chai-http
- create tests folder
- write tests for create user endpoint.
#### How should this be manually tested?
- clone the repository
- run `npm install` on the terminal
- create .env to setup db configuration
- run `db:migrate`
- run `npm test`  
#### Any background context you want to provide?
N/A
#### What are the relevant pivotal tracker stories?
#160451493

### CODE REVIEW

**Comment by kampkelly** [comment]
LGTM

**Review by slimsolz — COMMENTED** [review]
LGTM

**Comment by onwuvic** [comment]
LGTM

File: package.json [hunk 1]
```diff
@@ -7,13 +7,15 @@
     "babel-node": "babel-node --presets=env",
     "dev": "nodemon --exec npm run babel-node index.js",
     "start": "node index.js",
-    "test": "echo \"Error: no test specified\" && exit 1"
+    "test": "cross-env process.env.NODE_ENV=test mocha --compilers js:babel-core/register ./server/tests/**/'*.test.js' --recursive --timeout 50000 --exit || true",
```
Comment (s-akinrele) [inline package.json]: Please remove `--timeout`, can always put this there when we need it. 
We don't have a use case at the moment.

**Pull request merged**